test-loop-tests/src/utils/resharding.rs

```diff
@@ -852,3 +854,203 @@ pub(crate) fn check_state_cleanup(
+                        receiver_account.clone(),
+                        &signer,
+                        0,
+                        "call_yield_create_return_promise".to_string(),
+                        yield_payload.clone(),
+                        300 * TGAS,
+                        tip.last_block_hash,
+                    );
+                    store_and_submit_tx(
+                        &node_datas,
+                        &client_account_id,
+                        &txs,
+                        &signer_account,
+                        &receiver_account,
+                        tip.height,
+                        tx,
+                    );
+                    sent_flag.map(|flag| flag.set(true));
+                    tracing::debug!(target: "test", height=tip.height, ?signer_account, ?receiver_account, "sent promise yield tx");
+                };
+
+            // Function to retrieve the promise yield indices from the trie. This bypasses any other
+            // intermediate layer (caching, memtrie, flat-storage).
+            let get_promise_yield_indices_for_shard = |shard_uid| {
+                client_actor
+                    .client
+                    .chain
+                    .get_chunk_extra(&tip.prev_block_hash, &shard_uid)
+                    .ok()
+                    .map(|chunk_extra| {
+                        near_store::get_promise_yield_indices(&Trie::new(
+                            Arc::new(TrieDBStorage::new(
+                                TrieStoreAdapter::new(
+                                    client_actor.client.runtime_adapter.store().clone(),
+                                ),
+                                shard_uid,
+                            )),
+                            *chunk_extra.state_root(),
+                            None,
+                        ))
+                    })
+            };
+
+            // Run this action only once at every block height.
+            if latest_height.get() == tip.height {
+                return;
+            }
+            latest_height.set(tip.height);
+
+            let indices_shard_6 =
+                get_promise_yield_indices_for_shard(ShardUId { version: 3, shard_id: 6 });
+            let indices_shard_7 =
+                get_promise_yield_indices_for_shard(ShardUId { version: 3, shard_id: 7 });
+            let indices_shard_8 =
+                get_promise_yield_indices_for_shard(ShardUId { version: 3, shard_id: 8 });
+
+            tracing::debug!(target: "test", height=tip.height, epoch=?tip.epoch_id, ?indices_shard_6, ?indices_shard_7, ?indices_shard_8, "promise yield indices");
+
+            // At any height, if the shard exists and it is tracked, the promise yield indices trie node must exist.
+            assert_matches!(indices_shard_6, Some(Ok(_)) | None);
+            assert_matches!(indices_shard_7, Some(Ok(_)) | None);
+            assert_matches!(indices_shard_8, Some(Ok(_)) | None);
```

Comment:
nit: could we derive these, e.g., from `left_child_account`, `right_child_account`, and the shard layout? Feel free to ignore it if it's too much hassle, I'm fine with leaving it as is.

Reply:
Done